-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/try.scala:29:32 ------------------------------------------
29 |    (x: CanThrow[Exception]) => () => raise(new Exception)(using x)  // error
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                                Found:    {x} () => Nothing
   |                                Required: () => Nothing

longer explanation available when compiling with `-explain`
-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/try.scala:36:6 -------------------------------------------
36 |      () => // error
   |      ^
   |      Found:    {x} () => Int
   |      Required: () => Int
37 |        raise(new Exception)(using x)
38 |        22

longer explanation available when compiling with `-explain`
-- [E007] Type Mismatch Error: tests/neg-custom-args/captures/try.scala:47:4 -------------------------------------------
47 |    () => // error
   |    ^
   |    Found:    {x} () => Int
   |    Required: () => Int
48 |      raise(new Exception)(using x)
49 |      22

longer explanation available when compiling with `-explain`
-- Error: tests/neg-custom-args/captures/try.scala:22:28 ---------------------------------------------------------------
22 |  val a = handle[Exception, CanThrow[Exception]] { // error
   |                            ^^^^^^^^^^^^^^^^^^^
   |                            type argument is not allowed to capture the universal capability (* : Any)
